contracts/carrot-app/src/handlers/query.rs

```diff
@@ -79,94 +85,99 @@ fn query_compound_status(deps: Deps, env: Env, app: &App) -> AppResult<CompoundS
     })
 }
 
-fn query_position(deps: Deps) -> AppResult<PositionResponse> {
-    let position = POSITION.may_load(deps.storage)?;
+pub fn query_strategy(deps: Deps) -> AppResult<StrategyResponse> {
+    let config = CONFIG.load(deps.storage)?;
+
+    Ok(StrategyResponse {
+        strategy: config.balance_strategy.into(),
+    })
+}
 
-    Ok(PositionResponse { position })
+pub fn query_strategy_status(deps: Deps, app: &App) -> AppResult<StrategyResponse> {
+    let config = CONFIG.load(deps.storage)?;
+
+    Ok(StrategyResponse {
+        strategy: config
+            .balance_strategy
+            .query_current_status(deps, app)?
+            .into(),
+    })
 }
 
 fn query_config(deps: Deps) -> AppResult<Config> {
     Ok(CONFIG.load(deps.storage)?)
 }
-fn query_balance(deps: Deps, _app: &App) -> AppResult<AssetsBalanceResponse> {
-    let pool = get_osmosis_position(deps)?;
 
-    let balances = try_proto_to_cosmwasm_coins(vec![pool.asset0.unwrap(), pool.asset1.unwrap()])?;
-    let liquidity = pool.position.unwrap().liquidity.replace('.', "");
+pub fn query_balance(deps: Deps, app: &App) -> AppResult<AssetsBalanceResponse> {
+    let mut funds = Coins::default();
+    let mut total_value = Uint128::zero();
+
+    let config = CONFIG.load(deps.storage)?;
+    config.balance_strategy.0.iter().try_for_each(|s| {
+        let deposit_value = s
+            .yield_source
+            .ty
+            .user_deposit(deps, app)
+            .unwrap_or_default();
+        for fund in deposit_value {
+            let exchange_rate = query_exchange_rate(deps, fund.denom.clone(), app)?;
+            funds.add(fund.clone())?;
+            total_value += fund.amount * exchange_rate;
+        }
+        Ok::<_, AppError>(())
+    })?;
+
     Ok(AssetsBalanceResponse {
-        balances,
-        liquidity,
+        balances: funds.into(),
+        total_value,
     })
 }
 
-fn query_rewards(deps: Deps, _app: &App) -> AppResult<AvailableRewardsResponse> {
-    let pool = get_osmosis_position(deps)?;
+fn query_rewards(deps: Deps, app: &App) -> AppResult<AvailableRewardsResponse> {
```

Comment:
Would be nice to add pagination here as well

Reply:
Same as above